File: Problem_set_1/Introductory_problems/p08_Two_Sets/twoSets.py
num = int(input())

set1 = ""
set2 = ""

total_sum = num * (num + 1) // 2

if total_sum % 2 == 0:
    print("YES")

    if num % 4 == 3:
        set1 += "1 2 "
        count1 = 2

        set2 += "3 "
        count2 = 1

        i = 4
        while i <= num:
            set1 += str(i) + " " + str(i + 3) + " "
            count1 += 2

            set2 += str(i + 1) + " " + str(i + 2) + " "
            count2 += 2

            i += 4

        print(count2)
        print(set2)
        print(count1)
        print(set1)

    else:
        count = 0

        i = 1
        while i <= num:
            set1 += str(i) + " " + str(i + 3) + " "
            set2 += str(i + 1) + " " + str(i + 2) + " "
            count += 4
            i += 4

        print(count // 2)
        print(set1)
        print(count // 2)
        print(set2)

else:
    print("NO")

File: Problem_set_1/Introductory_problems/p08_Two_Sets/twoSets.ts
const input = require("prompt-sync")();

const num: bigint = BigInt(input());

let set1 = "";
let set2 = "";const sum = (num * (num + 1n)) / 2n;

if (sum % 2n === 0n) {
    console.log("YES");

    if (num % 4n === 3n) {
        set1 += "1 2 ";
        let count1 = 2;

        set2 += "3 ";
        let count2 = 1;

        for (let i = 4n; i <= num; i += 4n) {
            set1 += `${i} ${i + 3n} `;
            count1 += 2;

            set2 += `${i + 1n} ${i + 2n} `;
            count2 += 2;
        }

        console.log(count2);
        console.log(set2);
        console.log(count1);
        console.log(set1);
    } else {
        let count = 0;

        for (let i = 1n; i <= num; i += 4n) {
            set1 += `${i} ${i + 3n} `;
            set2 += `${i + 1n} ${i + 2n} `;
            count += 4;
        }

        console.log(count / 2);
        console.log(set1);
        console.log(count / 2);
        console.log(set2);
    }
} else {
    console.log("NO");
}